pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.
+            args = [
+                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
+                for a in args
+            ]
+
         # construct the tape
         self.construct(args, kwargs)
 
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        if list(res.shape) == [1]:
+            # HOTFIX: to maintain compatibility with core, we squeeze
+            # any output that has shape (1,).
+
+            # Get the namespace associated with the return type
+            res_type_namespace = res.__class__.__module__.split(".")[0]
+
+            if res_type_namespace in ("pennylane", "autograd"):
+                # For PennyLane and autograd we must branch, since
+                # 'squeeze' does not exist in the top-level of the namespace
+                return anp.squeeze(res)
+
+            return __import__(res_type_namespace).squeeze(res)
+
+        return res

```

Comment:
What is our roadmap to removing these hotfixes and changing the qnode behaviour?

Reply:
It depends on when we want to break autograd behaviour :) in the ADR, we punted this off as a 'future decision'.